scheduler-server/scheduling-dsl-compiler/src/libs/scheduler-ast.ts

```diff
@@ -1,3 +1,4 @@
+import "./TemporalPolyfillTypes.js";
```

Comment:
This import shouldn't be necessary.

I assume it's because the file itself complains about type errors without it - which is fine since we aren't running it in the environment that typescript thinks we're running it in.

We can replace this with a triple-slash directive (a special annotation typescript understands for local type inclusion, but does not translate into runtime code). https://www.typescriptlang.org/docs/handbook/triple-slash-directives.html#-reference-path-

Also probably not worth our effort if it doesn't immediately work.

Reply:
This `/// <reference path="./TemporalPolyfillTypes.js"; />`? File complains. Stops complaining when replacing `.js` with `.ts`.

Reply:
That makes sense to me. Typescript is looking for TS files, not JS files, and the whole extension proxying thing likely isn't build into the references yes. The whole extension proxying is terrible and caused by the terrible migration of module systems in Nodejs.